internal/project/project.go

```diff
@@ -1,241 +1,232 @@
 package project
 
 import (
-	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"net/http"
 	"os"
-	"os/exec"
-	"regexp"
-	"sort"
+	"path"
+	"path/filepath"
 	"strings"
-	"sync"
+	"time"
 
-	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/format/gitignore"
 	"github.com/pkg/errors"
+	"github.com/stateful/runme/internal/document"
+	"github.com/stateful/runme/internal/renderer/cmark"
+	"github.com/stateful/runme/internal/runner"
 )
 
-type Resolver struct {
-	cwd string
-
-	once sync.Once
-	repo *repo
-}
-
-func NewResolver(dir string) *Resolver {
-	return &Resolver{cwd: dir}
-}
-
-func (r *Resolver) openRepo() *repo {
-	r.once.Do(func() {
-		gitRepo, err := git.PlainOpen(r.cwd)
-		r.repo = &repo{Repository: gitRepo, err: err}
-	})
-	return r.repo
-}
-
 type Project struct {
+	RootDir    string
 	BranchName string
 	Commit     string
 	URL        string
+	matcher    gitignore.Matcher
 }
 
-func (r *Resolver) Get() (p Project, err error) {
-	repo, err := r.openRepo().Clone()
+func New(cwd string) (p Project) {
+	r := &Resolver{cwd: cwd}
+	p.RootDir = r.RootDir()
+
+	_, err := r.Get()
 	if err != nil {
-		return p, err
+		return p
 	}
 
-	p.BranchName = repo.BranchName()
-	p.Commit = repo.Commit()
-	p.URL = repo.URL()
+	w, err := r.repo.Worktree()
+	if err != nil {
+		return p
+	}
 
-	return p, repo.Err()
-}
+	patterns, err := gitignore.ReadPatterns(w.Filesystem, nil)
+	if err != nil {
+		return p
+	}
 
-type repo struct {
-	*git.Repository
-	err error
+	p.matcher = gitignore.NewMatcher(patterns)
+	return p
 }
 
-func (r *repo) Clone() (*repo, error) {
-	if err := r.Err(); err != nil {
-		return nil, err
+func (p *Project) isExcluded(file string, isDir bool) bool {
+	if p.matcher == nil {
+		return false
 	}
-	return &repo{Repository: r.Repository}, nil
+	return p.matcher.Match(strings.Split(file, "/"), isDir)
 }
 
-func (r *repo) Err() error { return r.err }
-
-func (r *repo) BranchName() string {
-	if r.Err() != nil {
-		return ""
-	}
-
-	ref, err := r.Head()
+func (p *Project) getAllMarkdownFiles() ([]string, error) {
+	root, err := os.Getwd()
 	if err != nil {
-		r.err = err
-		return ""
+		return nil, err
 	}
 
-	if ref.Name().IsBranch() {
-		return ref.Name().Short()
+	fsys := os.DirFS(root)
+	matches, err := fs.Glob(fsys, "**/*.md")
```

Comment:
This [does not work](https://github.com/golang/go/issues/11862) as expected, and in particular will ignore files in the root directory.

You should probably use a `WalkDir`-based solution instead.